### ISSUE TITLE
Encoding nested arrays, with strings that contain commas, breaks serialisation

### ISSUE THREAD

**Issue by assertchris** [livewire/livewire#587]
**Describe the bug**

I have this object:

```php
[
    [
        'id' => 'monthly',
        'name' => 'Monthly',
        'description' => 'Monthly access with smaller payments',
        'amount' => 10,
        'currency' => 'USD',
        'interval' => 'month',
        'canChange' => ['amount', 'currency'],
    ],
    [
        'id' => 'yearly',
        'name' => 'Yearly',
        'description' => 'Yearly access with large payments',
        'amount' => 100,
        'currency' => 'USD',
        'interval' => 'year',
        'canChange' => ['amount', 'currency'],
    ],
    ...$defaults,
];
```

I can assign that to a public property, and use it in the view, and emit it as an argument to an event. When I add a comma (to any of those description strings); the serialisation breaks such that I cannot emit the public property as an argument to an event.

```php
[
    [
        'id' => 'monthly',
        'name' => 'Monthly',
        'description' => 'Monthly access, with smaller payments',
        'amount' => 10,
        'currency' => 'USD',
        'interval' => 'month',
        'canChange' => ['amount', 'currency'],
    ],
    [
        'id' => 'yearly',
        'name' => 'Yearly',
        'description' => 'Yearly access, with large payments',
        'amount' => 100,
        'currency' => 'USD',
        'interval' => 'year',
        'canChange' => ['amount', 'currency'],
    ],
    ...$defaults,
];
```

```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class CustomizePlans extends Component
{
    public $type;
    public $plans;
    public $currentPlan;

    public function mount($type, $plans, $currentPlan = null)
    {
        $this->type = $type;
        $this->plans = $plans;
        $this->currentPlan = $currentPlan;
    }

    public function render()
    {
        $type = $this->type;
        $plans = $this->plans;
        $currentPlan = $this->currentPlan;

        return view('livewire.customize-plans', compact('type', 'plans', 'currentPlan'));
    }
}
```

```html
<div class="flex flex-col w-full">
    <h1 class="text-5xl font-bold leading-none mb-8">Who gets what?</h1>
    @foreach($plans as $plan)
        <button
            wire:click="$emit('onSelectedPlan', '{{ json_encode($plan) }}')"
            wire:key="{{ $type['id'] }}-{{$plan['id'] }}"
            class="
                relative flex flex-col w-full pl-8 py-2 mb-4
                @if($currentPlan and $currentPlan['id'] === $plan['id'])
                    text-gray-900
                @else
                    text-gray-500
                @endif
            "
        >   <div
                class="
                    absolute left-0 top-0 bottom-0 rounded
                    @if($currentPlan and $currentPlan['id'] === $plan['id'])
                        bg-indigo-500
                    @else
                        bg-gray-300
                    @endif
                "
                style="width: 5px"
            >&nbsp</div>
            <div class="text-xl font-semibold">{{ $plan['name'] }}</div>
            <div class="text-sm">{{ $plan['description'] }}</div>
        </button>
    @endforeach
</div>
```

**Expected behavior**
I expect to be able to encode/decode ad infinitum, without serialisation issues.

**Comment by AEK-BKF** [comment]
I think it's because your object id is string not qn integer, 
`@if($currentPlan and $currentPlan['id'] === $plan['id'])`
Try to put id as integer, hope it works.

**Comment by assertchris** [comment]
@AEK-BKF I'm not sure that's related. the commas are in the description field (which causes this issue), not the id field. To be clear: I have other components which define these plans, and they have string ids, and those don't exhibit the same behaviour. Indeed, when the descriptions don't contain commas then the string id is also fine.

**Comment by nuernbergerA** [comment]
I think this could be solved by #571 latest regex

**Comment by assertchris** [comment]
@nuernbergerA yeah, seems like it...

**Comment by nuernbergerA** [comment]
@assertchris should be fixed with [v0.7.3](https://github.com/livewire/livewire/releases/tag/v0.7.3)

**Comment by puntogestioncl** [comment]
Hello,

This error appears to still be present. When entering a string in a field, with a comma, it returns an error of :

`Undefined array key 3`

If more than one comma is added, it throws error

`SQLSTATE[42S22]: Column not found: 1054 Unknown column ' des de' in 'where clause'`

The text entered as test "ghasto, des de, deasdd"

I'm using

"laravel/framework": "^9.19",
"livewire/livewire": "^2.10",

A thousand apologies for my English, I'm using Google Translate
